### PR TITLE
Naval Patrol Station Trait (from that PR DrDiasyl made but went stale)

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I'm PR'ing an addition from #86619 (conflicts resolved and nits fixed), which unfortunately staled out despite being a fairly sound idea, and the author, DrDiasyl, hasn't been around for a while. Now, copypasting the description from the original PR:

>This very rare station trait (1 point less rare as Station-Wide Background Checks) does not allow any exterior antagonists like Nuclear Operatives, Wizards, Space Dragons, Aliens, etc to spawn, the only exception is Blob Infection.
>
>Internal threats like Traitors, Changelings, Heretic, Cults, Revolutionaries, etc can still spawn while the Station Trait is active.

Rare rounds with this threat would be refreshing for Crew and Security as they have no stress of getting randomly under heavy fire in the middle of the round, without just being a Green Start/White Dwarf with 0 threats.
>
>Paranoia and investigations will be more plausible when the Station Trait is active.

## Changelog
:cl: DrDiasyl aka DrTuxedo
add: NEW STATION TRAIT "GAMEMODE": Naval Patrol! The station sector is being crossed by a Naval fleet patrol, due to this NO external threats or infestations can board the station. But internal threats remain, so stay vigilant and paranoid without getting distracted...
/:cl:

### CODE REVIEW

**Comment by ElGitificador** [comment]
so would this mean midround threat budget is integrated into roundstart one or that midrounds budget will roll only for latejoins? and if the later is the case wouldnt that end up in a lot of unused trhreat?

**Comment by Ghommie** [comment]
There is more than a handful of midrounds and latejoin that can still roll. I'm unsure if this will be a recurrent probem but I'm leaning into "no" here.

**Comment by SomeoneYouProbablyKnow** [comment]
Word for word what i said on the last PR.

> Does this meaningfully change the round? Knowing i can't trust the other crewmen fully is a pretty natural state of the game. It's not much different from a black orbit where you really don't wanna trust anyone. That's why the background check thing works so well because it subverts that natural state.

Do hope Diasyl's doing fine, their PRs are very cool

**Comment by Ghommie** [comment]
I wish I had a scrying orb to tell you, but I can't say for sure if this reinforces paranoia when many players aren't informed of the station traits when half of them lack visibility. It's a good point, and I suppose it would benefit from some visual cues to make it, like several tiny spaceships lingering in the parallax background perhaps. That if I feel like spriting them.

**Comment by SomeoneYouProbablyKnow** [comment]
My main argument is that a normal round is one where you can't trust your colleagues, a round with this trait is a round where the trait makes it so you can't trust your colleagues. Not much changes in how you approach it even if you're aware of the trait. You still have reasons to powergame in case cult tries to eat your face, you can still trust metaprotected security, there's no ticking bomb forcing you to snitch or be more paranoid except general fear for your wellbeaing which is present in normal rounds

**Comment by SomeoneYouProbablyKnow** [comment]
Meaningful impact yes, the impact will be the round won't have those antags. The texture of the round will be different, there will be more traitors and fewer pirates. But i've seen dynamic do that all on its own.

I do not see a sociological impact because the internal logic for player behavior remain unchanged. The "background checks" trait has an impact on player behavior not just what spawns and what doesnt because it gives players definitive information that they can trust other crewmen (it's the same mechanism that defines a lot of sec crew interaction, meta-info that lets you know that player can't be an antag). And that makes players act differently because they have different information to act ON.

Knowing there's no nuckies or space dragons won't change how people play. Those antags are already the kind you respond to WHEN and IF they happen. I'm still left not trusting other crewmen like i was before the trait.

**Comment by zoomachina** [comment]
> Knowing there's no nuckies or space dragons won't change how people play. Those antags are already the kind you respond to WHEN and IF they happen. I'm still left not trusting other crewmen like i was before the trait.

Knowing there are no nukies or wizards in the round will certainly impact my gameplay

**Comment by SomeoneYouProbablyKnow** [comment]
You can disagree with me but if all you're gonna say is "nu huh" then we can just agree to disagree.

**Comment by Loafin34** [comment]
so, let's paint a picture. Midnight sun, 10 heretics, 3 security officers, one SSD, and a particularly bloodthirsty deputized assistant. This sounds like torture.

**Comment by DrDiasyl** [comment]
> so, let's paint a picture. Midnight sun, 10 heretics, 3 security officers, one SSD, and a particularly bloodthirsty deputized assistant. This sounds like torture.

Well without this Trait this would've been just 7 heretics + blob/dragon, I think Midnight Sun is really the keyword in your argument.
Also yes I'm alive just been off TG for a long time, I won't retake this PR though I'm fine with Ghommie having it.

File: code/modules/shuttle/battlecruiser_starfury.dm [hunk 1]
```diff
@@ -178,4 +178,12 @@
 					source = spawner,
 				)
 
-	priority_announce("Unidentified armed ship detected near the station.")
+	// If we have a Station Trait "Naval Patrol" it will not prevent the Lone Ops but just announce it presence
+	// You should've secured that disk! Not even Navy will stop a disk-sniffing-determined Lone Operative
```
Comment (DrDiasyl) [inline code/modules/shuttle/battlecruiser_starfury.dm:182]: The comment still mentions Lone Ops even though this is a announcement about a battlecruiser.

Comment (Ghommie) [reply code/modules/shuttle/battlecruiser_starfury.dm:182]: Oh thank you

**Comment by Time-Green** [comment]
I don't really like this as a station trait. I don't see it being that much of a deviation from a normal round? Crew-antagonist are our main antagonist, and getting rid of the back-up dancers doesn't change the experience that much

I also love killing puppies

**Pull request closed without merging**